### ISSUE TITLE
Check wheter is solved or not

### ISSUE THREAD

**Issue by KRK224** [javaparser/javaparser#4071]

**Comment by KRK224** [comment]
Hello, I'm a new user.

I've faced a same problem as [#1992](https://github.com/javaparser/javaparser/issues/1992).

I want to resolve only the MethodCallExpr in my source project, so I pass the ParserConfiguration with JavaParserTypeSolver as below

```java
package com.javaparser;

import java.io.File;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.List;

import com.github.javaparser.ParseResult;
import com.github.javaparser.ParserConfiguration;
import com.github.javaparser.ast.CompilationUnit;
import com.github.javaparser.ast.expr.MethodCallExpr;
import com.github.javaparser.resolution.UnsolvedSymbolException;

import com.github.javaparser.resolution.model.SymbolReference;
import com.github.javaparser.symbolsolver.utils.SymbolSolverCollectionStrategy;
import com.github.javaparser.utils.ProjectRoot;
import com.github.javaparser.utils.SourceRoot;

import java.util.Optional;
import com.github.javaparser.symbolsolver.JavaSymbolSolver;
import com.github.javaparser.symbolsolver.resolution.typesolvers.JavaParserTypeSolver;

/**
 * Hello world!
 *
 */
public class App {

  private static final String SRC_PATH = "src/main/resources/java-baseball/src/main/java";

  public static void main(String[] args) throws Exception {

    JavaParserTypeSolver javaParserTypeSolver = new JavaParserTypeSolver(new File(SRC_PATH));

    JavaSymbolSolver symbolSolver = new JavaSymbolSolver(javaParserTypeSolver);
    ParserConfiguration myConfiguration = new ParserConfiguration();
    myConfiguration.setSymbolResolver(symbolSolver);
    SymbolSolverCollectionStrategy symbolSolverCollectionStrategy = new SymbolSolverCollectionStrategy(myConfiguration);

    Path root = Paths.get(SRC_PATH);

    ProjectRoot projectRoot = symbolSolverCollectionStrategy.collect(root);

    List<SourceRoot> sourceRootList = projectRoot.getSourceRoots();

    for (SourceRoot sr : sourceRootList) {
      System.out.println("sourceRoot iterator:" + sr);

      List<ParseResult<CompilationUnit>> parseResults = sr.tryToParse();
      System.out.println("compilationUnit List size: " + parseResults.size());

      for (ParseResult<CompilationUnit> parseResult : parseResults) {
        try {
          Optional<CompilationUnit> optionalCompilationUnit = parseResult.getResult();
          if (optionalCompilationUnit.isPresent()) {
            CompilationUnit cu = optionalCompilationUnit.get();
            System.out.println("cu is:::" + cu.getStorage().get().getFileName());

            cu.findAll(MethodCallExpr.class).forEach(node -> {
              System.out.println("* Method Call::: " + node);
              // reflection type MethodCallExpr throwing UnsolvedSymbolException
              System.out.println(node.resolve().getQualifiedSignature());

              SymbolReference symbolReference = javaParserTypeSolver.tryToSolveType(node.getNameAsString());

              if (symbolReference.isSolved()) {
                System.out.println(node.resolve().getQualifiedSignature());
              }

            });

          }
        } catch (UnsolvedSymbolException e) {
          System.out.println("**** Got Error \n" + e.getMessage());
        }

      }

    }

  }
}


```

it resolved only method in my project well, but the problem is reflection type MethodCallExpr throwing UnsolvedSymbolException and stop the execution.

I've tried to use symbolReference.isSolved method but it didn't work. is there any solution to jump the failed solving or check whether is solved?

**Comment by jlerbsc** [comment]
I'm not sure I understand your problem because the error message is missing from the issue description. However, it could be a parser configuration problem. Below is a code snippet that allows you to configure the parser with multiple solvers.

    JavaParserTypeSolver javaParserTypeSolver = new JavaParserTypeSolver(new File(projectSrcFolder));
    TypeSolver reflectionTypeSolver = new ReflectionTypeSolver();

    CombinedTypeSolver combinedTypeSolver = new CombinedTypeSolver();
    combinedTypeSolver.add(reflectionTypeSolver);
    combinedTypeSolver.add(javaParserTypeSolver);
    
    JavaSymbolSolver symbolSolver = new JavaSymbolSolver(combinedTypeSolver);
    JavaParser.getStaticConfiguration().setSymbolResolver(symbolSolver);

**Comment by KRK224** [comment]
Thank you for your quick feedback:)

Setting the configuration to parser reflection type can be solution to the reflection type, but the problem is resolve() method  throw the exception when the methodCallExpr is not resolved.

so as [#1992 ](https://github.com/javaparser/javaparser/issues/1992) said, some maven/gradle project's dependencies are not present when their MethodCallExprs are parsing and it throws exception. In my case, what I want to resolve is only method and class that I wrote in my project internally, not the reflection type. so I remove the ReflectionTypeSolver from parser configuration.

In summary, I wonder how to:

1. judge whether a method call expression is resolved before the execution resolve() method.
2. findAll and filtering only the MethodCallExpr I wrote in my project internally.

**Comment by jlerbsc** [comment]
@ftomassetti gives the explanation on the origin of the exception UnsolvedSymbolException in https://github.com/javaparser/javaparser/issues/1992 . You should handle this exception in your code if you already know that the symbol solver is not configured to resolve certain classes.
